### ISSUE TITLE
numfig-format does not seem to honor ``{number}`` and ``{name}``

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3118]
I tried things like

```
numfig = True

numfig_format = {
    #'figure': 'Figure {number}',
    'table':  'Table {number} {name}',
    #'code-block': 'Listing {number}',
    #'section':    'Section {number}',
}
```

but never managed it to work. But the documentation says at http://www.sphinx-doc.org/en/master/config.html#confval-numfig_format : 

```
 numfig_format

    A dictionary mapping 'figure', 'table', 'code-block' and 'section' to strings that are used for format of figure numbers. As a special character, %s and {number} will be replaced to figure number. {name} will be replaced to figure caption.

    Default is to use 'Fig. %s' for 'figure', 'Table %s' for 'table', 'Listing %s' for 'code-block' and 'Section' for 'section'.
```

**Comment by tk0miya** [comment]
Ah, it's my mistake.
Only `:numref:` role allows both `{number}` and `{name}`. But `numfig_format` disallows.

`numfig_format` is used for the default link caption of `:numref:`. And it is also used for the prefix of captions of sections, figures, tables and code-blocks.
#2916 only modifies the format of link captions. It is not related with `numfig_format`.

I will fix it later. Thank you for letting me know :-)

**Comment by tk0miya** [comment]
Fixed at 7c19a2f
